src/actions/controller.ts

```diff
@@ -1,47 +1,61 @@
-import * as vscode from "vscode";
+import { extensions } from "vscode";
 import { getLogger } from "../logger/logger";
 import { IAction } from "./interfaces";
 import { _performAction } from "./performer";
 import { getParameter } from '../apis/parameters';
-import { forEach, uniq, get, split, compact } from "lodash";
+import { ActionsFactory } from './actionsFactory';
+import * as _ from "lodash";
+import * as actionsConfig from './actionsConfig';
+
 
 export class ActionsController {
   private static readonly actions: IAction[] = [];
 
-  public static loadActions() {
-    vscode.extensions.all.forEach(extension => {
-      const extActions = get(extension, "packageJSON.BASContributes.actions", []);
-      extActions.forEach((action: IAction) => {
-        ActionsController.actions.push(action);
+  public static loadContributedActions() {
+    const logger = getLogger().getChildLogger({ label: "loadContributedActions" });
+    extensions.all.forEach(extension => {
+      const extensionActions = _.get(extension, "packageJSON.BASContributes.actions", []);
```

Comment:
you can probably replace the `_.get(extension, "packageJSON.BASContributes.actions", []);` pattern
with modern javascript
```javascript
extension?.packageJSON?.BASContributes?.actions ?? []
```